### PR TITLE
test(FilterBar): rewrite test using react-testing-library

### PR DESCRIPTION
**Pull request by sidram05**
transition from enzyme to rect-testing-library

**Thank you for your contribution!** 👏

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-webcomponents-react/blob/master/CONTRIBUTING.md#contribute-code) section 
- [x] [Correct commit message style](https://github.com/SAP/ui5-webcomponents-react/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

**Comment by sidram05** [comment]
Hi @MarcusNotheis,

Please could you have a look 

Thanks
Sid

File: packages/main/src/components/FilterBar/FilterBar.test.tsx [hunk 1]
```diff
@@ -269,14 +269,14 @@ describe('FilterBar', () => {
...
+    const filterItemsFB = asFragment.find(FilterGroupItem);
+    const openFiltersDialogBtn = asFragment.find(Toolbar).find(Button).at(3);
     act(() => {
       openFiltersDialogBtn.prop('onClick')();
     });
-    wrapper.update();
+    asFragment.update();
 
-    const filterItemsDialog = wrapper.find('ui5-dialog').find(FilterGroupItem);
+    const filterItemsDialog = asFragment.find('ui5-dialog').find(FilterGroupItem);
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterBar/FilterBar.test.tsx:279]: This whole thing can be removed and replaced with:
```js
    fireEvent.click(screen.getByText('Filters'));

    const [selectFilterBar, selectFilterDialog] = screen.getAllByText('SELECT');
    expect(selectFilterDialog).toHaveAttribute('required', 'true');
    expect(selectFilterDialog.parentNode.parentNode.parentNode.parentNode.querySelector('ui5-checkbox')).toHaveAttribute('disabled', 'true');
```

In the end, that test should look like:
```js
it('Group Filter Items mounted in Dialog', () => {
    render(
      // ... leave the render content as it is
    );

    fireEvent.click(screen.getByText('Filters'));

    const [selectFilterBar, selectFilterDialog] = screen.getAllByText('SELECT');
    expect(selectFilterDialog).toHaveAttribute('required', 'true');
    expect(selectFilterDialog.parentNode.parentNode.parentNode.parentNode.querySelector('ui5-checkbox')).toHaveAttribute('disabled', 'true');
  });
```

**Review by MarcusNotheis — CHANGES_REQUESTED** [review]

**Review by MarcusNotheis — APPROVED** [review]
Hey @sidram05,
that was the last test using `enzyme` - thank you very much for your efforts to help us migrating towards `react-testing-library`, that really helped us a lot :) 

Best regards,
Marcus

**Comment by sidram05** [comment]
> Hey @sidram05,
> that was the last test using `enzyme` - thank you very much for your efforts to help us migrating towards `react-testing-library`, that really helped us a lot :)
> 
> Best regards,
> Marcus

Hi @MarcusNotheis, 

Thanks for all the help and guidance !
This my first OSS contribution.
Was a pleasure working with you.

Regards
Siddarth

**Pull request merged**